src/f32/sse2/vec3a.rs

```diff
@@ -0,0 +1,1098 @@
+    ///
+    /// When `s` is `0.0`, the result will be equal to `self`.  When `s` is `1.0`, the result
+    /// will be equal to `rhs`. When `s` is outside of range `[0, 1]`, the result is linearly
+    /// extrapolated.
+    #[doc(alias = "mix")]
+    #[inline]
+    pub fn lerp(self, rhs: Self, s: f32) -> Self {
+        self + ((rhs - self) * s)
+    }
+
+    /// Returns true if the absolute difference of all elements between `self` and `rhs` is
+    /// less than or equal to `max_abs_diff`.
+    ///
+    /// This can be used to compare if two vectors contain similar elements. It works best when
+    /// comparing with a known value. The `max_abs_diff` that should be used used depends on
+    /// the values being compared against.
+    ///
+    /// For more see
+    /// [comparing floating point numbers](https://randomascii.wordpress.com/2012/02/25/comparing-floating-point-numbers-2012-edition/).
+    #[inline]
+    pub fn abs_diff_eq(self, rhs: Self, max_abs_diff: f32) -> bool {
+        self.sub(rhs).abs().cmple(Self::splat(max_abs_diff)).all()
+    }
+
+    /// Returns a vector with a length no less than `min` and no more than `max`
+    ///
+    /// # Panics
+    ///
+    /// Will panic if `min` is greater than `max` when `glam_assert` is enabled.
+    #[inline]
+    pub fn clamp_length(self, min: f32, max: f32) -> Self {
+        glam_assert!(min <= max);
+        let length_sq = self.length_squared();
+        if length_sq < min * min {
+            self * (length_sq.sqrt().recip() * min)
+        } else if length_sq > max * max {
+            self * (length_sq.sqrt().recip() * max)
+        } else {
+            self
+        }
+    }
+
+    /// Returns a vector with a length no more than `max`
+    pub fn clamp_length_max(self, max: f32) -> Self {
+        let length_sq = self.length_squared();
+        if length_sq > max * max {
+            self * (length_sq.sqrt().recip() * max)
+        } else {
+            self
+        }
+    }
+
+    /// Returns a vector with a length no less than `min`
+    pub fn clamp_length_min(self, min: f32) -> Self {
+        let length_sq = self.length_squared();
+        if length_sq < min * min {
+            self * (length_sq.sqrt().recip() * min)
+        } else {
+            self
+        }
+    }
+
+    /// Fused multiply-add. Computes `(self * a) + b` element-wise with only one rounding
+    /// error, yielding a more accurate result than an unfused multiply-add.
+    ///
+    /// Using `mul_add` *may* be more performant than an unfused multiply-add if the target
+    /// architecture has a dedicated fma CPU instruction. However, this is not always true,
+    /// and will be heavily dependant on designing algorithms with specific target hardware in
+    /// mind.
+    #[inline]
+    pub fn mul_add(self, a: Self, b: Self) -> Self {
+        #[cfg(target_feature = "fma")]
+        unsafe {
+            _mm_fmadd_ps(self, b, c)
```

Comment:
Getting a compiler error in the `0.21` release here because `c` isn't defined :)